### PR TITLE
Add an ocaml-version option, format punned labelled arguments when OCaml 4.14+ syntax is selected

### PR DESCRIPTION
**Pull request by gpetiot**
Follow-up on #1756

### CODE REVIEW

**Review by nojb — COMMENTED** [review]
I think the version should be 4.14+, not 4.13+.

**Comment by gpetiot** [comment]
Ah indeed, I thought it would be included in the 4.13 release

**Comment by nojb** [comment]
> Ah indeed, I thought it would be included in the 4.13 release

I don't know anything about it, but I don't think it is at the moment.

**Comment by hhugo** [comment]
@gpetiot, what do you think about happening an issue for listing all syntaxes that need to be conditional on the ocaml version (e.g. (type a b))

**Comment by gpetiot** [comment]
I will add a list to #633

File: lib/Conf.ml [hunk 1]
```diff
@@ -935,6 +944,16 @@ module Formatting = struct
       (fun conf x -> {conf with nested_match= x})
       (fun conf -> conf.nested_match)
 
+  let ocaml_version =
+    let docv = "V" in
+    let doc = "Version of OCaml syntax of the output." in
+    let default = Ocaml_version.sys_version in
+    let default_doc = "the currently installed OCaml version" in
```
Comment (jberdine) [inline lib/Conf.ml:951]: @gpetiot I think that this is not as clear as it could be. Wouldn't "the version of OCaml used to build OCamlFormat" be better?

Comment (gpetiot) [reply lib/Conf.ml:951]: Indeed, done in #1765

**Pull request merged**